rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs

```diff
@@ -651,3 +655,71 @@ fn test_disburse_neuron() {
         assert_eq!(balance_after_disburse, balance_before_disburse + staked_amount - DEFAULT_TRANSFER_FEE.get_e8s());
     });
 }
+
+#[test]
+fn test_list_known_neurons() {
+    let rt = Runtime::new().unwrap();
+    rt.block_on(async {
+        let env = RosettaTestingEnvironment::builder()
+            .with_initial_balances(
+                vec![(
+                    AccountIdentifier::from(TEST_IDENTITY.sender().unwrap()),
+                    // A hundred million ICP should be enough
+                    icp_ledger::Tokens::from_tokens(100_000_000).unwrap(),
+                )]
+                .into_iter()
+                .collect(),
+            )
+            .with_governance_canister()
+            .build()
+            .await;
+
+        // Stake the minimum amount 100 million e8s
+        let staked_amount = 100_000_000u64;
+
+        env.rosetta_client
+            .create_neuron(
+                env.network_identifier.clone(),
+                &(*TEST_IDENTITY).clone(),
+                RosettaCreateNeuronArgs::builder(staked_amount.into()).build(),
+            )
+            .await
+            .unwrap();
```

Comment:
How about creating another neuron also that is not set as a `KnownNeuron`, and verify that the endpoint only returns one of the two?

Reply:
I am not testing the endpoint functionality of the governance canister here. The KnownNeuron is a specific neuron setting that can be fetched by anyone. The governance canister is responsible for  deciding which neurons are publicly known and which are not. 
I am testing the functionality of ICP Rosetta Client that anyone (in this case the anonymous principal) can fetch information about a KnownNeuron. 
Testing the functionality of the governance canister is out of scope for this PR IMHO. What do you think?